packages/devtools/lib/src/main.dart

```diff
@@ -148,27 +152,39 @@ class PerfToolFramework extends Framework {
     }
   }
 
-  void _initHotReloadServiceListener() {
-    final hotReloadServiceName = registrations.hotReload.service;
-    serviceManager.hasRegisteredService(hotReloadServiceName,
-        (bool reloadServiceAvailable) {
-      if (reloadServiceAvailable) {
-        _buildReloadRestartButtons();
-      } else {
-        clearGlobalActions();
-      }
-    });
+  void _initHotReloadRestartServiceListeners() {
+    serviceManager.hasRegisteredService(
+      registrations.hotReload.service,
+      (bool reloadServiceAvailable) {
+        if (reloadServiceAvailable) {
+          _buildReloadButton();
+        } else {
+          removeGlobalAction(_reloadTooltip);
+        }
+      },
+    );
+
+    serviceManager.hasRegisteredService(
+      registrations.hotRestart.service,
+      (bool reloadServiceAvailable) {
+        if (reloadServiceAvailable) {
+          _buildRestartButton();
+        } else {
+          removeGlobalAction(_restartTooltip);
+        }
+      },
+    );
   }
 
-  void _buildReloadRestartButtons() async {
+  void _buildReloadButton() async {
     // TODO(devoncarew): We currently create hot reload events when hot reload
     // is initialed, and react to those events in the UI. Going forward, we'll
     // want to instead have flutter_tools fire hot reload events, and react to
     // them in the UI. That will mean that our UI will update appropriately
     // even when other clients (the CLI, and IDE) initial the hot reload.
 
     final ActionButton reloadAction =
-        ActionButton('icons/hot-reload-white@2x.png', 'Hot Reload');
+        ActionButton(FlutterIcons.hotReloadWhite_2x, _reloadTooltip);
```

Comment:
We do have a 1x version of the icon. Should I reference the 1x or the 2x version in this constructor?

Reply:
reference the regular version and you will automatically get the high resolution version as needed.